templates/serverpod_templates/projectname_server_upgrade/Dockerfile

```diff
@@ -1,29 +1,22 @@
-# If you update the dart version, make sure the image is
-# compatible with the busybox image.
 FROM dart:3.2.5 AS build
 
 WORKDIR /app
 COPY . .
 
 RUN dart pub get
-RUN dart compile exe bin/main.dart -o bin/main
+RUN dart compile exe bin/main.dart -o bin/server
 
-# If you update the busybox version, make sure the image is
-# compatible with the dart image.
-FROM busybox:1.36.1-glibc
-
-ENV runmode=development
-ENV serverid=default
-ENV logging=normal
-ENV role=monolith
+FROM scratch
 
 COPY --from=build /runtime/ /
-COPY --from=build /app/bin/main /app/bin/main
+COPY --from=build /app/bin/server server
 COPY --from=build /app/config/ config/
 COPY --from=build /app/web/ web/
 
+
 EXPOSE 8080
 EXPOSE 8081
 EXPOSE 8082
 
-CMD app/bin/main --mode $runmode --server-id $serverid --logging $logging --role $role
+# The default flags used when starting the server, can be manually overridden by passing flags to the docker run command
+ENTRYPOINT ["./server", "--mode=production", "--server-id=default", "--logging=normal", "--role=monolith"]
```

Comment:
Question (For my knowledge): Don't we required a CMD to be defined? How does it know that any flags should be passed to the `./server` binary?

Reply:
That is how entrypoint works, the difference between cmd and entrypoint is that you can append / modify flags on entrypoint and on cmd you have to replace the entire script instead.